pkg/vz/vm_darwin.go

```diff
@@ -0,0 +1,463 @@
+	ciDataPath := filepath.Join(driver.Instance.Dir, filenames.CIDataISO)
+	isBaseDiskCDROM, err := iso9660util.IsISO9660(baseDiskPath)
+	if err != nil {
+		return err
+	}
+	var configurations []vz.StorageDeviceConfiguration
+
+	if isBaseDiskCDROM {
+		baseDiskAttachment, err := vz.NewDiskImageStorageDeviceAttachment(baseDiskPath, true)
+		if err != nil {
+			return err
+		}
+		baseDisk, err := vz.NewUSBMassStorageDeviceConfiguration(baseDiskAttachment)
+		if err != nil {
+			return err
+		}
+		configurations = append(configurations, baseDisk)
+	}
+
+	ciDataAttachment, err := vz.NewDiskImageStorageDeviceAttachment(ciDataPath, true)
+	if err != nil {
+		return err
+	}
+	ciData, err := vz.NewVirtioBlockDeviceConfiguration(ciDataAttachment)
+	if err != nil {
+		return err
+	}
+	configurations = append(configurations, ciData)
+
+	diffDiskAttachment, err := vz.NewDiskImageStorageDeviceAttachment(diffDiskPath, false)
+	if err != nil {
+		return err
+	}
+
+	diffDisk, err := vz.NewVirtioBlockDeviceConfiguration(diffDiskAttachment)
+	if err != nil {
+		return err
+	}
+	configurations = append(configurations, diffDisk)
+
+	vmConfig.SetStorageDevicesVirtualMachineConfiguration(configurations)
+	return nil
+}
+
+func attachDisplay(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	if driver.Yaml.Video.Display != nil {
+		graphicsDeviceConfiguration, err := vz.NewVirtioGraphicsDeviceConfiguration()
+		if err != nil {
+			return err
+		}
+		scanoutConfiguration, err := vz.NewVirtioGraphicsScanoutConfiguration(1920, 1200)
+		if err != nil {
+			return err
+		}
+		graphicsDeviceConfiguration.SetScanouts(scanoutConfiguration)
+
+		vmConfig.SetGraphicsDevicesVirtualMachineConfiguration([]vz.GraphicsDeviceConfiguration{
+			graphicsDeviceConfiguration,
+		})
+	}
+	return nil
+}
+
+func attachConsole(_ *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	consoleDevice, err := vz.NewVirtioConsoleDeviceConfiguration()
+	if err != nil {
+		return err
+	}
+	spiceAgentAttachment, err := vz.NewSpiceAgentPortAttachment()
+	if err != nil {
+		return err
+	}
+	spiceAgentName, err := vz.SpiceAgentPortAttachmentName()
+	if err != nil {
+		return err
+	}
+	spiceAgentPort, err := vz.NewVirtioConsolePortConfiguration(
+		vz.WithVirtioConsolePortConfigurationAttachment(spiceAgentAttachment),
+		vz.WithVirtioConsolePortConfigurationName(spiceAgentName),
+	)
+	if err != nil {
+		return err
+	}
+
+	consoleDevice.SetVirtioConsolePortConfiguration(0, spiceAgentPort)
+	vmConfig.SetConsoleDevicesVirtualMachineConfiguration([]vz.ConsoleDeviceConfiguration{
+		consoleDevice,
+	})
+	return nil
+}
+
+func attachFolderMounts(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	mounts := make([]vz.DirectorySharingDeviceConfiguration, len(driver.Yaml.Mounts))
+	for i, mount := range driver.Yaml.Mounts {
+		expandedPath, err := localpathutil.Expand(mount.Location)
+		if err != nil {
+			return err
+		}
```

Comment:
Seems to need mkdir here, LGTM other than that, thanks

Reply:
Done the changes and updated the vz template to include /tmp/lima